### PR TITLE
Remove listen `changeDate` after destroy DateRangePicker

### PR DESCRIPTION
**Pull request by lusever**
Need for correct recreate DateRangePicker on same inputs.

### CODE REVIEW

File: js/bootstrap-datepicker.js [hunk 1]
```diff
@@ -1567,6 +1567,7 @@
 		},
 		destroy: function(){
 			$.map(this.pickers, function(p){ p.destroy(); });
+			$(this.inputs).off('changeDate', this.dateUpdated);
```
Comment (vsn4ik) [inline js/bootstrap-datepicker.js]: `this.dateUpdated` !== `$.proxy(this.dateUpdated, this);`

Comment (lusever) [inline js/bootstrap-datepicker.js]: Yes. But it is work for jQuery. From [docs](http://api.jquery.com/off/):

> When jQuery attaches an event handler, it assigns a unique id to the handler function. Handlers proxied by jQuery.proxy() or a similar mechanism will all have the same unique id (the proxy function), so passing proxied handlers to .off may remove more handlers than intended. In those situations it is better to attach and remove event handlers using namespaces.

Comment (lusever) [inline js/bootstrap-datepicker.js]: И чего?

Comment (vsn4ik) [inline js/bootstrap-datepicker.js]: https://github.com/eternicode/bootstrap-datepicker/blob/master/js/bootstrap-datepicker.js#L2001 используется нэймспейс. Я считаю что лучше использовать нэймспейс, даже jQuery советует.

Comment (lusever) [inline js/bootstrap-datepicker.js]: I changed the way of removing events to namespace.

**Comment by vsn4ik** [comment]
You need to add namespace here (https://github.com/eternicode/bootstrap-datepicker/blob/master/js/bootstrap-datepicker.js#L1513) and add testcase.

**Comment by lusever** [comment]
I added namespace.
@vsn4ik prompt me please place where to create the testcase correctly.

**Comment by acrobat** [comment]
@vsn4ik I think we can't add a namespace now, this would mean a BC break. Why is the namespace necessary ?

**Comment by vsn4ik** [comment]
@acrobat Let's roll out without tests. Namespace is not necessary.

**Comment by acrobat** [comment]
@lusever Can you remove the `.datepicker` namespace on both events? If that still fixes your issue the pr is good to go!

**Comment by lusever** [comment]
@acrobat I removed namespaces.

**Comment by lusever** [comment]
If I use the code which is currently:
```javascript
$('.inputs input').on('changeDate', myFunc)
$('.inputs').datepicker({inputs: $('input')})
$('.inputs').data('datepicker').remove(); // removed myFunc too
```
I think the original version is more correct. It is like in `_unapplyEvents`. 
I think it is necessary to restore the original version. Then `myFunc` will not be removed.

**Comment by acrobat** [comment]
So the changes in this PR are not correct @lusever?

**Comment by lusever** [comment]
@acrobat now is correct.

**Comment by acrobat** [comment]
Thanks @lusever!

**Pull request merged**